docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,197 @@
+ensure the same level of High Availability Prometheus has i.e. can lose all
+replicas but one.
+
+*Log Querying*
+
+Get all logs for a given period, node(s), pod regex, limited list of predefined
+labels and free keywords text.
+Typical Zenko use case: collecting all logs across several components, related
+to a S3 uniq request.
+
+*Log statistics (nice to have)*
+
+The Log centralisation service also offer the ability to consume statistics
+about the logs like the number of occurrences of one type of log during a
+certain period of time.
+
+*Monitorable/Observable service (health, performances and alerts)*
+
+The Platform Administrator can monitor capacity usage, IOPS, latency and
+bandwidth of the Log centralisation service. He can also monitor the health of
+the service (i.e. if some active alerts exist). He can visualize active and
+past alerts impacting the service.
+
+*Performances (TBD)*
+
+Typical workloads can generate around 1000 logs per second per node.
+
+User Stories
+------------
+
+- As a Platform Administrator, I want to browse all MetalK8s containers logs
+  (from all servers) from a unique endpoint, in order to ease distributed
+  k8s and metalk8s services error investigation.
+
+- As a Platform Administrator, I want to browse non container (kubelet,
+  containerd, salt-minion, cron) logs (from all servers) from a
+  unique endpoint, in order to ease System error investigation.
+
+- As a Storage Administrator, I want to browse all Solution instance containers
+  logs (from all servers) from a unique endpoint, in order to ease Solution
+  instance Error investigation.
+
+- As a Platform Administrator, I want to push all containers logs to an
+  external log centralisation system, In Order to archive it or aggregate it
+  with other application logs.
+
+
+(some other US extracted from Loki design doc)
+
+- After receiving an alert on my service and drilling into the query associated
+  with said alert, I want to quickly see the logs associated with the jobs
+  which produced those timeseries at the time of the alert.
+
+- After a pod or node disappears, I want to be able to retrieve logs from just
+  before it died, so I can diagnose why it died.
+
+- After discovering an ongoing issue with my service, I want to extract a
+  metric from some logs and combine it with my existing time series data.
+
+- I have a legacy job which does not expose metrics about errors - it only logs
+  them.  I want to build an alert based on the rate of occurrences of errors in
+  the log.
+
+
+Deployment & Configuration
+--------------------------
+
+The log centralisation service is scheduled on infra nodes.
+A platform Administrator can operate the service as follow:
+
+- add persistent storage
+- configure max retention period
+- adjust the number of replicas
+- push logs to an external log centralisation service
+- configure alerting related to log centralisation service
```

Comment:
What configuration? Thresholds/silences/... (the usual)? Define new alerts? For monitoring what, the log service or the app(s) from which we are collecting logs?

Reply:
add details